woocommerce/woocommerce-ios
# REST API: Remove XMLRPC check for site address login

Closes: #8770
Please also review https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/736.


## Description

This PR removes the XMLRPC requirement to handle site credential login. Changes include:
- Added a new use case `SiteCredentialLoginUseCase`. This use case works similarly to `SiteCredentialViewModel`:
  - All cookies are cleared before handling login. This is important since existing cookies might cause the login to succeed even with incorrect site credentials.
  - Use the input credentials to handle cookie-nonce authentication.
  - Trigger a call to WP.org REST API to check if the authentication is successful. Here I'm using the plugin endpoint since it has already been implemented with cookie authentication.
  - The `SiteCredentialViewModel` will need to be updated to use this use case, but I'll handle that in a separate PR to avoid bloating this PR.
- Applied a new config `skipXMLRPCCheckForSiteAddressLogin` for the authenticator and implemented the delegate method `handleSiteCredentialLogin` to trigger `SiteCredentialLoginUseCase`.

Note: at the time of writing, we are forced to update WPKit version due to the requirement from the latest version of WPAuthenticator. Please follow the updates in an internal thread: p1675142099386409-slack-CC7L49W13.

## Testing instructions

- Update `enableSiteAddressLoginOnly` in `AuthenticationManager` to `true` to force the site address login only experience and build the app.
- On the prologue screen, select Enter your site address and proceed with the address of a self-hosted store.
- Enter incorrect site credentials - notice that an error is displayed for the incorrect credentials.
- Enter the correct credentials - the login should succeed and the app should navigate to the home screen.
- Notice in Xcode console that all Tracks event for the login flow should be sufficient for both success and failure cases, and no mention of XMLRPC should be present.

Please feel free to test again with `enableSiteAddressLoginOnly` disabled. Site address login should still use XMLRPC and succeed with sites that have working XMLRPC urls.

## Screenshots

N/A
---
